src/serac/physics/thermomechanics.hpp

```diff
@@ -390,7 +390,7 @@ class Thermomechanics : public BasePhysics {
   template <typename MaterialType, typename StateType = Empty>
   void setMaterial(MaterialType material, std::shared_ptr<QuadratureData<StateType>> qdata = EmptyQData)
```

Comment:
A much simpler design would be to pass material by a const reference. std::move and std::forward are to be used sparingly and with much more discipline.